src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-button.component.ts

```diff
@@ -29,6 +30,7 @@ export class ProgrammingAssessmentRepoExportButtonComponent {
     @Input() participationIdList: number[];
     @Input() participantIdentifierList: string; // comma separated
     @Input() singleParticipantMode = false;
+    @Input() selectedProgrammingExercises: ProgrammingExercise[];
```

Comment:
It seams that now both the exerciseID and the exercise array are necessary inputs. If they are optional I would at least add a ? behind the variable names.

Since you pass the array of programming exercises here, would it be possible to get rid of the exerciseId attribute and only pass one exercise inside an array in the place where the exerciseId is used?

Reply:
I now removed the `exerciseId` and only left the exercise array. The `length` of the array is now used to conditionally switch between download of a single and multiple exercises 👍